### DISCUSSION TITLE
Rook keeps taking over external drive and formatting it when it isn't empty

### DISCUSSION THREAD

**Discussion by 2fst4u** [rook/rook#12963] [Q&A]
As per the docs only drives that are empty should be taken over by rook, but I have just lost data (nothing too important, just annoyed) on an external drive that I was intending to use for other purposes.

I don't have any filtering on drive usage in my config, because I do want all empty drives to be used for rook. It's just that this drive wasn't empty, it had files on it.

I don't have full logs, but I'm just wondering if there is something else I'm missing that I was expecting it to do?

Or alternatively, can the filtering of drives in the cluster config set a blacklist across all nodes for one particular drive uuid? If I can do this then I know this drive will be safe from being erased.

**Comment by travisn** [comment]
Rook is expected to skip devices or partitions that have a filesystem as described in the [prerequisites](https://rook.io/docs/rook/latest/Getting-Started/Prerequisites/prerequisites/#ceph-prerequisites). Does your external drive not show that it has a filesystem with `lsblk`?

The filter-related properties in the CephCluster CR are currently only for inclusion, not for exclusion.

**Reply by 2fst4u** [reply]
It was showing an ext4 filesystem and it had files on the device that could be viewed and browsed. I can't understand why it happened.

**Reply by travisn** [reply]
Any chance you have the osd prepare pod log from when the OSD was mistakenly created? The log would show what command rook ran and perhaps show why it didn't detect the filesystem.

**Reply by 2fst4u** [reply]
No I think a few of those of those pods have come and gone now sorry. I've had to use a filter to prevent it from happening again.

Can the filter be used for one device only while having all other devices grab and new drives automatically as normal? Or if one device is filtered do all devices need to be explicitly declared?

**Reply by travisn** [reply]
If the devices are explicitly declared in the list, the `deviceFilter` wouldn't be used. The `deviceFilter` is used at the same time as `useAllDevices: true`. So if you can find a filter that includes your desired devices, but not the external device. Note you can test the regular expressions used for the filter on a site like [this](https://regex101.com/), just make sure to select Golang as the type of filter.

**Reply by 2fst4u** [reply]
Ok so I have to name every device in the cluster if I want to add one? That should be ok.
